spec/observables/from-callback-spec.js

```diff
@@ -0,0 +1,93 @@
+/* globals describe, it, expect */
+var Rx = require('../../dist/cjs/Rx');
+var Observable = Rx.Observable;
+
+describe('Observable.fromCallback', function () {
+  it('should emit one value from a callback', function (done) {
+    function callback (datum, cb) {
+      cb(datum);
+    }
+    var cbToObs = Observable.fromCallback(callback);
+
+    cbToObs(42)
+      .subscribe(function (x) {
+        expect(x).toBe(42);
+      }, null,
+      function (x) {
+        expect(x).toBe(undefined);
+        done();
+      });
+  });
+
+  it('should emit one value chosen by a selector', function (done) {
+    function callback (datum, cb) {
+      cb(null, datum);
+    }
+    var cbToObs = Observable.fromCallback(callback, null, function (err, datum) { return datum; });
+
+    cbToObs(42)
+      .subscribe(function (x) {
+        expect(x).toBe(42);
+      }, null,
+      function (x) {
+        expect(x).toBe(undefined);
+        done();
+      });
+  });
+
+  it('should override `this` in the callback', function (done) {
+    function callback (cb) {
+      cb(this.value);
+    }
+    var cbToObs = Observable.fromCallback(callback, {value: 42});
+
+    cbToObs()
+      .subscribe(function (x) {
+        expect(x).toBe(42);
+      }, null,
+      function (x) {
+        expect(x).toBe(undefined);
+        done();
+      });
+  });
+
+  it('should emit an error when the selector throws', function (done) {
+    function callback (cb) {
+      cb(42);
+    }
+    var cbToObs = Observable.fromCallback(callback, null, function (err) { throw new Error('Yikes!'); });
+
+    cbToObs()
+      .subscribe(null,
+      function (err) {
+        expect(err.message).toBe('Yikes!');
+        done();
+      },
+      null
+      );
+  });
+
+  it('should not emit, throw or complete if immediately unsubscribed', function (done) {
+    var nextSpy = jasmine.createSpy('next');
+    var throwSpy = jasmine.createSpy('throw');
+    var completeSpy = jasmine.createSpy('complete');
+    var timeout;
+    function callback (datum, cb) {
+      timeout = setTimeout(function () {
+        cb(datum);
+      });
+    }
+    var subscription = Observable.fromCallback(callback)(42)
+      .subscribe(nextSpy, throwSpy, completeSpy);
+    subscription.unsubscribe();
+
+    setTimeout(function () {
+      expect(nextSpy).not.toHaveBeenCalled();
+      expect(throwSpy).not.toHaveBeenCalled();
+      expect(completeSpy).not.toHaveBeenCalled();
+
+      clearTimeout(timeout);
+      done();
+    });
+  });
```

Comment:
I guess for now this async test is ok. When we have Subscriber `start()` feature, we can rework this test.